### ISSUE TITLE
Possibly a bug? Im unsure sorry if incorrect in advance.

### ISSUE THREAD

**Issue by pallebone** [opnsense/core#5095]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

YES - [ ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
YES - [ ] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

From forum post where nobody replied here: https://forum.opnsense.org/index.php?topic=23900.0

After changing to wireguard kernel version I noticed:
On my firewall I have a  pppoe connection and that MTU on my pppoe interface was 1492. On wireguard interface the MTU was set to 1420 which would be acceptable on an 1500 wan interface setup. Because my pppoe is 1492 it caused issues because the WG mtu should be 1412 in this scenario.

However I found it was impossible to change the MTU on the WG interface. I set 1412 as the MTU on my wireguard interface and it rebooted the Firewall but found even after reboot the overview area showed MTU of 1420 still on the WG interface. IE the change on the interface did not actually help.

This was causing some timeouts on speedtest on my phone and also WG to run slow.

So I had to go to Firewall - Settings - Normalization and set a Max mss of 1412. This resolved the issues and brought wireguard back to normal.

I did not experience these issues on the WG-go implementation. Somehow it worked without these changes fine.
Also I dont understand why setting an MTU on the WG interface did not change the MTU after a reboot.

Is this a bug or normal? I dont know. Sorry in advance. If it is not a bug please close.

I can post more info if required. Also is there a command you can run from the cli to verify you are using the wireguard kmod version as opposed to wireguard go? From the gui I just see the WG-go service is stopped. I assume that means kernel version is being used.

Kind regards
Pete

Apologies if this ticket raised is wasting your time. Its not my intention to do so. I asked on the forum and nobody replied to me so thats why I am raising here.

**Comment by pallebone** [comment]
Just letting you know I updates to 21.7 and it still has the same "issue" on this version also.

**Comment by pallebone** [comment]
I found the answer to this problem by stumbling accross it when looking at something else.

The guides on opnsense site tell you to tick the box "Prevent interface removal". When this is ticked then setting an mtu on an interface (not just WG, any interface" seems to be ignored. If you uncheck this then change mtu it allows changing. 

Unsure why preventing interface removal checkbox is actually suggested as leaving unticked allows this to work and seems to have no undesired effects.

Closing as this was the solution.
